apps/web/src/pages/Teams/TeamMembersPage/components/RemoveTeamMemberDialog.tsx

```diff
@@ -0,0 +1,150 @@
+import { getFullNameLabel } from "~/utils/nameUtils";
+import { TeamMember } from "~/utils/teamUtils";
+import { toast } from "@gc-digital-talent/toast";
+
+interface RemoveTeamMemberDialogProps {
+  user: TeamMember;
+  team: Team;
+}
+
+const RemoveTeamMemberDialog = ({
+  user,
+  team,
+}: RemoveTeamMemberDialogProps) => {
+  const intl = useIntl();
+  const [{ fetching }, executeMutation] = useUpdateUserAsAdminMutation();
+
+  const [isOpen, setIsOpen] = React.useState<boolean>(false);
+
+  const handleRemove = async () => {
+    await executeMutation({
+      id: user.id,
+      user: {
+        roles: {
+          detach: {
+            roles: user.roles.map((role) => role.id),
+            team: team.id,
+          },
+        },
+      },
+    })
+      .then((res) => {
+        if (!res.error) {
+          setIsOpen(false);
+          toast.success(
+            intl.formatMessage({
+              defaultMessage: "Member removed from team successfully",
+              id: "guzGyX",
+              description:
+                "Alert displayed to user when a team member is removed",
+            }),
+          );
+        }
+      })
+      .catch(() => {
+        toast.error(
+          intl.formatMessage({
+            defaultMessage: "Member removed from team failed",
+            id: "7Mbt01",
+            description:
+              "Alert displayed to user when an error occurs while removing a team member",
+          }),
+        );
+      });
+  };
+
+  const userName = getFullNameLabel(user.firstName, user.lastName, intl);
+  const teamName = getLocalizedName(team.displayName, intl);
+
+  const label = intl.formatMessage({
+    defaultMessage: "Remove member",
+    id: "DyHeaZ",
+    description: "Label for the dialog to remove a users team membership",
+  });
+
+  return (
+    <Dialog.Root open={isOpen} onOpenChange={setIsOpen}>
+      <Dialog.Trigger>
+        <Button color="error" mode="outline" aria-label={label}>
+          <TrashIcon data-h2-height="base(x.75)" data-h2-width="base(x.75)" />
+        </Button>
+      </Dialog.Trigger>
+      <Dialog.Content>
+        <Dialog.Header>{label}</Dialog.Header>
+        <Dialog.Body>
+          <p data-h2-margin="base(x1, 0)">
+            {intl.formatMessage(
+              {
+                defaultMessage:
+                  "You are about to remove <strong>{userName}</strong> from the team <strong>{teamName}</strong>.",
+                id: "/esEW+",
+                description: "Help text for the remove team membership dialog",
+              },
+              {
+                userName,
+                teamName,
+              },
+            )}
+          </p>
+          {user.roles.length ? (
+            <>
+              <p data-h2-margin="base(x1, 0)">
+                {intl.formatMessage({
+                  defaultMessage:
+                    "They will love all of the following team roles:",
```

Comment:
they may love those roles, but in this case they are losing them

Reply:
Fixed in [fix typo](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5811/commits/8c867539e7d8c6ab5ad24b9ec5bc236e52d50e99)